addons/hw_drivers/tools/helpers.py

```diff
@@ -341,6 +341,22 @@ def load_certificate():
         start_nginx_server()
     return True
 
+def delete_iot_handlers(iot_handlers_path):
+    """
+    Delete all the drivers and interfaces
+    This is needed to avoid conflicts
+    with the newly downloaded drivers
+    """
+    try:
+        for directory in ['drivers', 'interfaces']:
+            directory_path = iot_handlers_path.joinpath(directory)
+            files = os.listdir(directory_path)
+            for file in files:
+                os.remove(directory_path.joinpath(file))
```

Comment:
If I understand this lines correctly, you remove every files in a given folder, is it correct ? It looks a bit convoluted and I guess it might break if a file happen to be a folder. Wouldn't it be better to just delete the folder ? I guess it will be recreated if missing, isn't it ?

Reply:
![image](https://github.com/odoo/odoo/assets/36443074/078a31aa-dc45-4b12-80b4-5007f30ce93e)

Reply:
at least maybe put a check in case customer have some custom codes which include folders ?

Reply:
Good idea, but it's unnecessary with the current code. The directories won't be listed through `list_files_by_os' so we won't try to delete them.Also, the files included in the subdirectories won't be listed neither. Anyhow, the `path.unlink` method is unable to delete a repository. So the custom code currently will be simply left alone